src/components/processor-factory/form-processor-factory.component.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import useProcessorDependencies from '../../hooks/useProcessorDependencies';
+import useInitialValues from '../../hooks/useInitialValues';
+import { FormRenderer } from '../renderer/form/form-renderer.component';
+import { type FormSchema } from '../../types';
+import { CustomHooksRenderer } from '../renderer/custom-hooks-renderer.component';
+import { useFormFields } from '../../hooks/useFormFields';
+import { useConcepts } from '../../hooks/useConcepts';
+import { useFormFieldValidators } from '../../hooks/useFormFieldValidators';
+import { type FormProcessorContextProps } from '../../types';
+import { useFormFieldsMeta } from '../../hooks/useFormFieldsMeta';
+import { useFormFactory } from '../../provider/form-factory-provider';
+import { useFormFieldValueAdapters } from '../../hooks/useFormFieldValueAdapters';
+import { EncounterFormProcessor } from '../../processors/encounter/encounter-form-processor';
+
+interface FormProcessorFactoryProps {
+  formJson: FormSchema;
+  isSubForm: boolean;
+}
+
+const FormProcessorFactory = ({ formJson, isSubForm = false }: FormProcessorFactoryProps) => {
+  const { patient, sessionMode, formProcessors, layoutType, location, provider, sessionDate, visit } = useFormFactory();
+
+  // TODO: load processor from the registry
+  const processor = useMemo(() => {
+    const ProcessorClass = formProcessors[formJson.processor];
+    if (processor) {
+      return processor;
+    }
+    if (ProcessorClass) {
+      return new ProcessorClass(formJson);
+    }
+    console.error(`Form processor ${formJson.processor} not found, defaulting to EncounterFormProcessor`);
+    return new EncounterFormProcessor(formJson);
+  }, [formProcessors, formJson.processor]);
+
+  const [processorContext, setProcessorContext] = useState<FormProcessorContextProps>({
+    patient,
+    formJson,
+    sessionMode,
+    layoutType,
+    location,
+    currentProvider: provider,
+    processor,
+    sessionDate,
+    visit,
+  });
+  const { formFields: rawFormFields, conceptReferences } = useFormFields(formJson);
+  const { concepts: formFieldsConcepts, isLoading: isLoadingConcepts } = useConcepts(conceptReferences);
```

Comment:
`isLoading` here seems to be unused?

Reply:
I'd also add that this was one of the sort of requests I'd hoped we could avoid with the o3forms endpoint, but if that doesn't work for some reason, let me know.

Reply:
I agree that loading all the concepts is an expensive operation, and we should rely on backend optimization. I've tried loading a form using the O3 forms endpoint, but in my experience, it doesn't seem to resolve missing labels (by inferring them from the concept).

For us to fully depend on the O3 forms endpoint, we need to add support for resolving subforms.

Reply:
> it doesn't seem to resolve missing labels

Ah, yes, it doesn't magically create `label` elements. Basically, this was so, in the NGX engine, we could override the label with translations or, absent those, the concept name, since the concept name isn't always the appropriate string. That said, the endpoint should give you an element called "conceptReferences" which has, for every concept in the form, the concept's UUID and display name.

I guess that's a good point about subforms, but you could always just request them as separate forms for now.

Reply:
Since the o3forms endpoints now supports subforms and correctly queries forms by name, I'm gonna create a ticket to track the work of fully leveraging this backend optimisation.